### ISSUE TITLE
Feature Request: Link plots to code line that made them

### ISSUE THREAD

**Issue by oscarrutt** [julia-vscode/julia-vscode#2503]
The Julia quasi-notebook feature is fantastic. However, it lacks a valuable feature of native notebooks--plots become divorced from the line of code that generated them. In a notebook, the plot is directly below the line or cell that made it. In the julia-vscode case, all of the plots live in a single tab and you can't tell which plot belongs to which line of code. The request is to make a link back from each plot to the generating line of code. This could momentarily highlight the line while the eye follows the action from one tab to the other, or the highlight could be persistent, until another plot is linked back. This could be made to happen automatically or upon some user interaction such as clicking something--Command-clicking the plot, for example (Mac key convention used here) and/or a menu item in the plot pane. There will be instances where the user does not want to have the line of code displayed, perhaps because she/he is editing some other line and doesn't want the code pane to scroll. [Edit--adding to original post] Of course the linking would also work with plots displayed in the Plot Navigator.

Also, the extension provides tiny icons at the end of some lines of code for various things. To complement the above suggestion, make a tiny icon next to a line that generates a plot; clicking the icon takes you to the appropriate plot in the plot tab. [Edit--adding to original post] An additional use of the tiny icon would be a hover action whereby hovering on the icon causes a rendering of the plot to appear. The rendered plot might be a  miniature and might leverage off of the recent addition of the Plot Navigator which also uses miniatures.

**Comment by pfitzseb** [comment]
Good ideas, although not trivial to implement.

**Comment by oscarrutt** [comment]
> Good ideas, although not trivial to implement.

True, I suppose. But speaking from complete ignorance here, you can click on an error line in the terminal and it jumps you to the offending line of code. Maybe the line number could be embedded with the plot somehow.
